lib/tasks/publishing_api.rake

```diff
@@ -292,6 +292,9 @@ namespace :publishing_api do
         edition.translations.pluck(:locale).each do |locale|
           puts "Fixing edition #{edition.content_id}, locale #{locale}"
           Services.publishing_api.discard_draft(edition.content_id, locale: locale)
+        rescue GdsApi::HTTPNotFound, GdsApi::HTTPUnprocessableEntity
```

Comment:
You might want to update the commit message to include error code 422 (in addition to 404).

I don't know if there's any possibility we'll ever get this error, but publishing-api can [return a 409](https://github.com/alphagov/publishing-api/blob/8b9b753866a013d012093f93a51c0809e2a74f96/app/commands/base_command.rb#L84), in addition to [404/422](https://github.com/alphagov/publishing-api/blob/8b9b753866a013d012093f93a51c0809e2a74f96/app/commands/v2/discard_draft.rb#L27).

Reply:
Thanks for finding that line of code. I did wonder why I was seeing a 404 on my first Integration test, and a 422 on the subsequent test 😅 I'll split that out into separate unit tests.

I won't worry about a 409 unless I come across it in the full Integration run here: https://deploy.integration.publishing.service.gov.uk/job/run-rake-task/220338/console